packages/db-base/lib/redisHandler.js

```diff
@@ -311,8 +381,64 @@ class RedisHandler extends EventEmitter {
      * @param arr Array to send out
      */
     sendArray(responseId, arr) {
+        if (this.multiActive && this.multiResponseIds.includes(responseId)) {
+            this._handleMultiResponse(responseId, Resp.encodeArray(this.encodeRespArray(arr)));
+            return;
+        }
+
         this.sendResponse(responseId, Resp.encodeArray(this.encodeRespArray(arr)));
     }
+
+    /**
+     * Handles a 'multi' command
+     *
+     * @param {number} responseId ID of the response
+     * @private
+     */
+    _handleMulti(responseId) {
+        if (this.multiActive) {
+            this.log.warn(`${this.socket} Conflicting multi call`);
+        }
+        this.sendString(responseId, 'OK');
+        this.multiActive = true;
+        this.execCalled = false;
+        this.multiResponseIds = [];
+        this.multiResponseCount = 0;
+        this.multiResponseObj = {};
+    }
+
+    /**
+     * Handles an 'exec' command
+     *
+     * @param {number} responseId ID of the response
+     * @private
+     */
+    _handleExec(responseId) {
+        this.execCalled = true;
+        this.execId = responseId;
+
+        // maybe we have all fullfilled yet
+        if (this.multiResponseCount === this.multiResponseIds.length) {
+            this.multiActive = false;
+            this._sendQueued(this.execId, Resp.encodeArray(Object.values(this.multiResponseObj)));
+        }
+    }
+
+    /**
+     * Handles a multi response
+     *
+     * @param {number} responseId ID of the response
+     * @param {Buffer} buf buffer to include in response
+     * @private
+     */
+    _handleMultiResponse(responseId, buf) {
+        this.multiResponseObj[responseId] = buf;
+        this.multiResponseCount++;
+        if (this.execCalled && this.multiResponseCount === this.multiResponseIds.length) {
+            this.multiActive = false;
+            this._sendQueued(this.execId, Resp.encodeArray(Object.values(this.multiResponseObj)));
```

Comment:
Are you sure that the order odf the "Object.values" is always "as inserted into the object"?

Reply:
was just an admin cache problem

Reply:
And maps are guaranteed order? ... good to know :-)

Reply:
Yes, see [javascript maps api](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#objects_vs._maps) -> Key Order

Reply:
Oh need to check.. entry insertion is not always correct if it is meant like that, because we are changing the entry later on in a not known way

Reply:
yes, works as expected:

```javascript
const map = new Map();

map.set('test', 'testing');
map.set('lest', 'lesting');
map.set('test', 'newTesting')

console.log(Array.from(map.values())); // [ 'newTesting', 'lesting' ]

for (const [key, value] of map) {
    console.log(key) // test, lest
}
```